python/ray/data/_internal/planner/exchange/sort_task_spec.py

```diff
@@ -162,13 +165,14 @@ def sample_boundaries(
         sample_results = [
             sample_block.remote(block, n_samples, sort_key) for block in blocks
         ]
-        sample_bar = ProgressBar(
-            SortTaskSpec.SORT_SAMPLE_SUB_PROGRESS_BAR_NAME,
-            len(sample_results),
-            unit="block",
-        )
+        if sample_bar is None:
+            sample_bar = ProgressBar(
+                SortTaskSpec.SORT_SAMPLE_SUB_PROGRESS_BAR_NAME,
+                len(sample_results),
+                unit="block",
```

Comment:
hmm, if we are setting the unit as `block` here, how come the screenshot in https://github.com/ray-project/ray/pull/47106#issuecomment-2291876295 shows the unit as `row` for the `Sort Sample` bars?

Reply:
oh i see, this is only for the case where the bar is not passed from calling `sample_boundaries`. understood